codec/src/main/java/io/netty/handler/codec/compression/Bzip2Encoder.java

```diff
@@ -123,17 +123,7 @@ protected void encode(ChannelHandlerContext ctx, ByteBuf in, ByteBuf out) throws
                     Bzip2BlockCompressor blockCompressor = this.blockCompressor;
                     final int length = in.readableBytes() < blockCompressor.availableSize() ?
                                     in.readableBytes() : blockCompressor.availableSize();
-                    final int offset;
-                    final byte[] array;
-                    if (in.hasArray()) {
-                        array = in.array();
-                        offset = in.arrayOffset() + in.readerIndex();
-                    } else {
-                        array = new byte[length];
-                        in.getBytes(in.readerIndex(), array);
-                        offset = 0;
-                    }
-                    final int bytesWritten = blockCompressor.write(array, offset, length);
+                    final int bytesWritten = blockCompressor.write(in, in.readerIndex(), length);
```

Comment:
Use intrinstics for the win ... `length` -> `Math.min(in.readableBytes(), blockCompressor.availableSize());`